mm/tlsf/mm_tlsf.c

```diff
@@ -1124,15 +1130,20 @@ FAR void *mm_realloc(FAR struct mm_heap_s *heap, FAR void *oldmem,
 {
   FAR void *newmem;
 
+  /* If oldmem is NULL, then realloc is equivalent to malloc */
+
   if (oldmem == NULL)
     {
       return mm_malloc(heap, size);
     }
 
-  if (size == 0)
+  /* If size is zero, reallocate to the minim size object, so
+   * the memory pointed by oldmem is freed
+   */
+
+  if (size < 1)
     {
-      mm_free(heap, oldmem);
-      return NULL;
+      size = 1;
```

Comment:
Passing size=0 and size=1 to mm_malloc have the same effect. A minimum chunk size is allocated. So I think the above can be removed.